### PR TITLE
Sarah/ev2

### PR DESCRIPTION
**Pull request by sarahpeiffer**
Ev2 deployment changes for agent release

### CODE REVIEW

File: deployment/agent-deployment/ServiceGroupRoot/RolloutSpecs/RolloutSpecs.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      "Name": "ContainerInsightsAgent",
+      "RolloutType": "Major",
+      "BuildSource": {
+        "Parameters": {
+          "VersionFile": "buildver.txt"
+        }
+      },
+      "Notification": {
+        "Email": {
+          "To": "sarahpeiffer@microsoft.com"
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/RolloutSpecs/RolloutSpecs.json:16]: please update to team dl

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+set -e
+
+# Note - This script used in the pipeline as inline script
+
+if [ -z $AGENT_IMAGE_TAG_SUFFIX ]; then
+  echo "-e error value of AGENT_IMAGE_TAG_SUFFIX variable shouldnt be empty"
+  exit 1
+fi
+
+if [ ${#AGENT_IMAGE_TAG_SUFFIX} -ne 8 ]; then
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:11]: lets not this validation as we use -hotfix, or -1 etc. for the hotfix release.

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  exit 1
+fi
+
+if [ -z $AGENT_RELEASE ]; then
+  echo "-e error AGENT_RELEASE shouldnt be empty"
+  exit 1
+fi
+
+#Download agentimage tarball from blob storage account
+echo "Downloading tarball image from $AGENT_IMAGE_URI"
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:22]: Downloading windows agent image tarball image from $AGENT_IMAGE_URI

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+if [ ! -f $AGENT_IMAGE_TAR_FILE_NAME ]; then
+    echo "Agent tarfile: ${AGENT_IMAGE_TAR_FILE_NAME} does not exist, unable to continue"
+    exit 1
+fi
+
+#Install crane
+echo "Installing crane"
+wget -O crane.tar.gz https://github.com/google/go-containerregistry/releases/download/v0.4.0/go-containerregistry_Linux_x86_64.tar.gz
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:33]: what if, downloading of the crane file broken, so check the status code of the wget or check existence of the crane.tar.g. if it exists, just fail the release.

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+
+#Login to az cli and authenticate to acr
+echo "Login cli using managed identity"
+az login --identity
+
+echo "Getting acr credentials"
+TOKEN_QUERY_RES=$(az acr login -n "$ACR_NAME" -t)
+TOKEN=$(echo "$TOKEN_QUERY_RES" | jq -r '.accessToken')
+DESTINATION_ACR=$(echo "$TOKEN_QUERY_RES" | jq -r '.loginServer')
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:45]: check if the TOKEN is empty, then fail the release with the appropriate error message.

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+./crane auth login "$DESTINATION_ACR" -u "00000000-0000-0000-0000-000000000000" -p "$TOKEN"
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:46]: check for $DESTINATION_ACR, if its empty throw an error and fail the release.

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+fi
+
+echo "Pushing file $TARBALL_IMAGE_FILE to $AGENT_IMAGE_FULL_PATH"
+./crane push *.tar "$AGENT_IMAGE_FULL_PATH"
+
+
+#Delete agentimage tarball from blob storage to prevent future conflicts
+echo "Deleting agentimage copy from blob storage"
+
+BLOB_EXIST_RESULT=$(az storage blob exists --container-name $STORAGE_CONTAINER_NAME --name $RELEASE_ID --account-name $STORAGE_ACCOUNT_NAME --sas-token $AGENT_IMAGE_SAS)
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:64]: you will have detect windows or linux, if its windows, win-$RELEASE_ID.

File: deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+BLOB_EXIST=$(echo "$BLOB_EXIST_RESULT" | jq -r '.exists')
+echo $BLOB_EXIST_RESULT
+echo $BLOB_EXIST
+
+if $BLOB_EXIST; then
+  az storage blob delete --container-name "${STORAGE_CONTAINER_NAME}" --name "${RELEASE_ID}" --account-name "${STORAGE_ACCOUNT_NAME}" --sas-token "${AGENT_IMAGE_SAS}"
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh:70]: **ReleaseId**=win-$RELEASE_ID to check blob existence and deletion.

File: deployment/agent-deployment/ServiceGroupRoot/Parameters/ContainerInsights.Linux.Parameters.json [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            "path": "artifacts.tar.gz"
+          }
+        },
+        "launch": {
+          "command": [
+            "/bin/bash",
+            "pushAgentToAcr.sh"
+          ],
+          "environmentVariables": [
+            {
```
Comment (ganga1980) [inline deployment/agent-deployment/ServiceGroupRoot/Parameters/ContainerInsights.Linux.Parameters.json:30]: I think, we can remove this "WINDOWS" environment variable from here, right? For the Windows ( in windows parameter file), you will be having this environment variable with value "win-". if the value of "WINDOWS" either empty or doesnt exist, determines that, its Linux, right?

Comment (sarahpeiffer) [reply deployment/agent-deployment/ServiceGroupRoot/Parameters/ContainerInsights.Linux.Parameters.json:30]: Yes--removing now

**Review by ganga1980 — APPROVED** [review]
lgtm. lets test it out.

**Pull request merged**